src/legacy/core_plugins/console/public/src/components/settings_modal.tsx

```diff
@@ -0,0 +1,263 @@
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiOverlayMask,
+  EuiSpacer,
+  EuiSwitch,
+  EuiText,
+} from '@elastic/eui';
+import { DevToolsSettings } from './dev_tools_settings';
+
+interface Props {
+  onSaveSettings: (newSettings: DevToolsSettings) => Promise<void>;
+  onClose: () => void;
+  refreshAutocompleteSettings: () => void;
+  settings: DevToolsSettings;
+}
+
+export function DevToolsSettingsModal(props: Props) {
+  const [fontSize, setFontSize] = useState(props.settings.fontSize);
+  const [wrapMode, setWrapMode] = useState(props.settings.wrapMode);
+  const [fields, setFields] = useState(props.settings.autocomplete.fields);
+  const [indices, setIndices] = useState(props.settings.autocomplete.indices);
+  const [templates, setTemplates] = useState(props.settings.autocomplete.templates);
+  const [polling, setPolling] = useState(props.settings.polling);
+  const [tripleQuotes, setTripleQuotes] = useState(props.settings.tripleQuotes);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    return () => {
+      // TODO
+    };
+  }, []);
+
+  function saveSettings() {
+    if (isLoading) {
+      return;
+    }
+    setIsLoading(true);
+    props.onSaveSettings({
+      fontSize,
+      wrapMode,
+      autocomplete: {
+        fields,
+        indices,
+        templates,
+      },
+      polling,
+      tripleQuotes,
+    });
+
+    setIsLoading(false);
+  }
+
+  return (
+    <EuiOverlayMask>
+      <EuiModal
+        data-test-subj="devToolsSettingsModal"
+        className="devToolsSettingsModal"
+        onClose={props.onClose}
+      >
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            <FormattedMessage id="console.settingsPage.pageTitle" defaultMessage="Settings" />
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+
+        <EuiModalBody className="euiText euiText--medium euiText--constrainedWidth">
+          <EuiFlexGrid columns={1}>
+            <EuiFlexItem>
+              <EuiFlexItem>
+                <EuiText>
+                  <p>
+                    <FormattedMessage
+                      id="console.settingsPage.fontSizeLabel"
+                      defaultMessage="Font Size"
+                    />
+                  </p>
+                </EuiText>
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiFieldNumber
+                  autoFocus
+                  data-test-subj="fontSizeInput"
+                  value={fontSize}
+                  onChange={e => {
+                    setFontSize((e.target.value as unknown) as number);
+                  }}
+                />
+              </EuiFlexItem>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiSwitch
+                checked={wrapMode}
+                data-test-subj="settingsWrapLines"
+                id="wrapLines"
+                label={
+                  <FormattedMessage
+                    defaultMessage="Wrap Long Lines"
```

Comment:
Watch your casing, we always use sentence case

```suggestion
                    defaultMessage="Wrap long lines"
```